src/main/java/graphql/nadel/result/ResultComplexityAggregator.java

```diff
@@ -4,12 +4,13 @@
 
 import java.util.LinkedHashMap;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 
 @Internal
 public class ResultComplexityAggregator {
     private int totalNodeCount = 0;
```

Comment:
Think this needs to be an `AtomicInteger` seeing as `serviceNodeCounts` has concurrency issues.

Reply:
yea thats a good idea, done now